core/tauri/src/menu/icon.rs

```diff
@@ -207,7 +207,7 @@ impl<R: Runtime> IconMenuItem<R> {
   /// - **Windows / Linux**: Unsupported.
   pub fn set_native_icon(&mut self, _icon: Option<NativeIcon>) -> crate::Result<()> {
     #[cfg(target_os = "macos")]
-    return run_main_thread!(self, |self_: Self| self_
+    return run_main_thread!(mut self, |self_: Self| self_
```

Comment:
`set_native_icon` shouldn't need to take a mutable reference, so you need to just change the method signature instead.

Reply:
This should be fixed now